pybliometrics/scopus/scopus_search.py

```diff
@@ -215,6 +217,179 @@ def get_eids(self):
+
+        bib_tex_type_article = "Article"
+        bib_tex_type_in_proceedings = "InProceedings"
+        bib_tex_type_in_collection = "InCollection"
+        bib_tex_type_book = "Book"
+        bib_tex_type_techreport = "TechReport"
+
+        bib_library = bibtexparser.Library()
+
+        results = self.results
+        
+        if results:
+            for result in results:
+                # print(result)
+
+                document_type = result.subtypeDescription
+                aggregation_type = result.aggregationType
+
+
+                # Item key
+                year = result.coverDate[0:4]
+
+                key_author: str = ""
+
+                author_names = result.author_names
+                
+                if author_names:
+                    key_author = author_names.split(",", 1)[0]
+                    
+                    if not imitate_scopus_export:
+                        # Remove potential white spaces
+                        key_author = "".join(key_author.split())
+                
+                key = "".join([key_author, year])
+
+                # Authors
+                authors = ""
+                if author_names:
+                    authors = " and ".join(author_names.split(";"))
+
+                # Pages
+                pages = None
+                page_range = result.pageRange
+                if page_range:
+                    pages = page_range.replace("-", " – ")
+
+                # Affiliation
+                affiliation: str = result.affilname
+                if affiliation:
+                    affiliation = "; ".join(affiliation.split(";"))
+
+                # Author keywords
+                author_keywords: str = result.authkeywords
+                if author_keywords:
+                    author_keywords = "; ".join(author_keywords.split(" | "))
+
+                # All information
+                bib_tex_type = None
+                if (document_type in [type_article, type_review, type_short_survey, type_editorial, type_note, type_letter, type_data_paper, type_erratum, type_conference_review, type_conference_paper, type_retracted, type_none] and aggregation_type == aggregation_type_journal) or (document_type in [type_article, type_review, type_short_survey, type_note] and aggregation_type == aggregation_type_trade_journal) or (document_type == type_article and aggregation_type == aggregation_type_none):
+                    bib_tex_type = bib_tex_type_article
+                if aggregation_type == aggregation_type_conference_proceedings or (document_type == type_conference_paper and aggregation_type in [aggregation_type_book, aggregation_type_none]):
+                    bib_tex_type = bib_tex_type_in_proceedings
+                elif aggregation_type == aggregation_type_book_series or (document_type in [type_book_chapter, type_article, type_editorial] and aggregation_type == aggregation_type_book):
+                    bib_tex_type = bib_tex_type_in_collection
```

Comment:
It would be more readable to define suitable doctypes as a set beforehand:
```
in_collection = {"ed", "ar", "ch"}
```
etc.

Reply:
Mh, you mean only for the subcheck `document_type in [...]`?